### PR TITLE
Add quaternionic package

### PR DESCRIPTION
**Pull request by moble**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/quaternionic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/{{ owner }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: c83bd808174621c38b96a103b14a772733cb70e874800170ae15bb132b439019
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<36]
```
Comment (scopatz) [inline recipes/quaternionic/meta.yaml:16]: This can be a noarch: python package

Comment (moble) [reply recipes/quaternionic/meta.yaml]: Sorry.  I always forget how to make it noarch but still limit the python versions.

File: recipes/quaternionic/meta.yaml [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 
 requirements:
   host:
     - python
-    - pip >=19.0
+    - pip >=20.0.2
     - poetry
   run:
-    - python
+    - python >=3.6
```
Comment (moble) [inline recipes/quaternionic/meta.yaml]: @scopatz Is this right?

Comment (scopatz) [reply recipes/quaternionic/meta.yaml]: both pythons should have the same pin, but other than that it looks right!

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**